### PR TITLE
Docker Image

### PR DESCRIPTION
**Pull request by JonZudell**
Simple docker image for running will include more changes as this is a draft PR

### CODE REVIEW

**Comment by JonZudell** [comment]
A Dockerfile has been added as well as an action to publish the image on semver tag.

The manifest docker action is now run in the published image resulting in 8x faster processing.

A script has been added to enable quick testing through `bash -c $(curl ...)`.

File: .github/workflows/manifest-check.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  manifest-check:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/jonzudell/manifest/manifest:${{ inputs.MANIFEST_IMAGE_TAG }}
```
Comment (BlakeWilliams) [inline .github/workflows/manifest-check.yml]:
```suggestion
      image: ghcr.io/blakewilliams/manifest/manifest:${{ inputs.MANIFEST_IMAGE_TAG }}
```

File: .github/workflows/manifest-check.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      options: --user 1001
```
Comment (BlakeWilliams) [inline .github/workflows/manifest-check.yml:20]: Why 1001 here?

File: .github/workflows/manifest-head.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  call-manifest-check-head:
+    uses: ./.github/workflows/manifest-check.yml
+    with:
+      MANIFEST_IMAGE_TAG: latest
```
Comment (BlakeWilliams) [inline .github/workflows/manifest-head.yml]: Can you fix up the newline endings on these files?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+RUN echo "#!/bin/sh" > /home/manifestuser/start_cmd.sh
+RUN echo "BRANCH=\${1:-\$(git rev-parse --abbrev-ref HEAD)}" >> /home/manifestuser/start_cmd.sh
+RUN echo "HASH=\${2:-\$(git rev-parse HEAD)}" >> /home/manifestuser/start_cmd.sh
+# RUN echo "git fetch origin \$BRANCH \$HASH" >> /home/manifestuser/start_cmd.sh
+# RUN echo "git diff origin/\$BRANCH...HEAD | manifest inspect --sha \$HASH --formatter github --strict" >> /home/manifestuser/start_cmd.sh
+# RUN chmod +x /home/manifestuser/start_cmd.sh
+# RUN echo "#!/bin/sh" > /home/manifestuser/start_cmd.sh
+# #RUN echo "git fetch origin \$BRANCH \$HASH" >> /home/manifestuser/start_cmd.sh
+# RUN echo "BRANCH=\$(git rev-parse --abbrev-ref HEAD)" >> /home/manifestuser/start_cmd.sh
+# RUN echo "HASH=\$(git rev-parse HEAD)" >> /home/manifestuser/start_cmd.sh
```
Comment (BlakeWilliams) [inline Dockerfile]:
```suggestion
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+WORKDIR /manifest
+COPY ./ ./
+
+RUN go install -buildvcs=false ./cmd/manifest
+ENV PATH="/home/manifestuser/go/bin:${PATH}"
+
+WORKDIR /app
+
+RUN git config --global --add safe.directory /app
+RUN echo "#!/bin/sh" > /home/manifestuser/start_cmd.sh
```
Comment (BlakeWilliams) [inline Dockerfile]: If we need this, maybe we turn this into `script/ci`?

Ideally we'll build this into the app, though.

File: README.md [hunk 6]
```diff
@@ -5,6 +5,9 @@ using configurable rules. It is language agnostic, passing the relevant pull
 request and diff information to scripts via JSON while using the resulting
 stdout JSON to comment on the PR/diffs, fail the build, etc.
 
+## Try Manifest
+Execute the following command from your repository
+`docker run -it -v $(pwd):/app -e MANIFEST_GITHUB_TOKEN=$MANIFEST_GITHUB_TOKEN ghcr.io/jonzudell/manifest/manifest:v0.0.10`
```
Comment (BlakeWilliams) [inline README.md]:
```suggestion
## Try Manifest

Execute the following command from your repository
`docker run -it -v $(pwd):/app -e MANIFEST_GITHUB_TOKEN=$MANIFEST_GITHUB_TOKEN ghcr.io/blakewilliams/manifest/manifest:v0.0.10`
```

I also wonder if this should be try manifest, or if we should add this as a way to install manifest in the instructions below?

**Review by BlakeWilliams — COMMENTED** [review]
I left a few comments that I think are worth addressing, but overall looks good!

File: .github/workflows/manifest-check.yml (hunk 2 above)
Comment (JonZudell) [reply .github/workflows/manifest-check.yml:20]: 1001 is the UID for manifestuser in the container
```
# Sets up manifestuser as user 1001 ready to execute manifest
RUN adduser -D -u 1001 manifestuser
USER manifestuser:1001
```

Without this git complains of dubious repo ownership when executing in actions.

File: Dockerfile (hunk 5 above)
Comment (JonZudell) [reply Dockerfile]: Good idea. I will look into building it into the app as part of a separate change set.

**Review by JonZudell — COMMENTED** [review]
Addressed the requested changes. I will address removing scripts/ci.sh in a second change set.

File: .github/workflows/manifest-check.yml (hunk 2 above)
Comment (BlakeWilliams) [reply .github/workflows/manifest-check.yml:20]: Hmmm, that's a little obtuse by possibly fine for now.

File: .github/workflows/manifest-check.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+          fetch-depth: 10000
+
+      - name: Fetch target branch
+        run: |
+          git fetch --depth=1 origin ${{ github.event.pull_request.base.ref }} \
+            ${{ github.event.pull_request.head.ref }}
+
+      - name: Manifest inspection
+        run: |
+          /home/manifestuser/start_cmd.sh ${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.ref }}
```
Comment (BlakeWilliams) [inline .github/workflows/manifest-check.yml]:
```suggestion
          /home/manifestuser/cli.sh ${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.ref }}
```

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR /manifest
+COPY ./ ./
+RUN go install -buildvcs=false ./cmd/manifest
+ENV PATH="/home/manifestuser/go/bin:${PATH}"
+
+# /app is where the repository to be checked should be mounted
+WORKDIR /app
+RUN git config --global --add safe.directory /app
+
+CMD /app/scripts/ci.sh
```
Comment (BlakeWilliams) [inline Dockerfile]:
```suggestion
CMD /app/scripts/ci.sh

```

File: scripts/ci.sh [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+BRANCH=${1:-$(git rev-parse --abbrev-ref HEAD)}
+HASH=${2:-$(git rev-parse HEAD)}
+
+git diff origin/$BRANCH...HEAD | manifest inspect --sha $HASH --formatter github
```
Comment (BlakeWilliams) [inline scripts/ci.sh]:
```suggestion
git diff origin/$BRANCH...HEAD | manifest inspect --sha $HASH --formatter github

```

File: README.md [hunk 10]
```diff
@@ -5,10 +5,12 @@ using configurable rules. It is language agnostic, passing the relevant pull
 request and diff information to scripts via JSON while using the resulting
 stdout JSON to comment on the PR/diffs, fail the build, etc.
 
-## Installing Manifest
+## Try Manifest
```
Comment (BlakeWilliams) [inline README.md]:
```suggestion
```

**Review by BlakeWilliams — APPROVED** [review]
LGTM, lets merge this in and see how it builds on the next PR

**Pull request merged**